src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -33,6 +34,13 @@
 #include <boost/algorithm/string/predicate.hpp>
 
 namespace ripple {
+
+extern std::tuple<std::shared_ptr<Ledger>, std::uint32_t, uint256>
```

Comment:
I'd prefer we don't use `extern` functions from cpp files (see previous comment on `extern`).